jsonrpc/types.go

```diff
@@ -45,67 +45,120 @@ func toTransaction(t *types.Transaction, b *types.Block, txIndex int) *transacti
-	LogsBloom    types.Bloom    `json:"logsBloom"`
-	Difficulty   argUint64      `json:"difficulty"`
-	Number       argUint64      `json:"number"`
-	GasLimit     argUint64      `json:"gasLimit"`
-	GasUsed      argUint64      `json:"gasUsed"`
-	Timestamp    argUint64      `json:"timestamp"`
-	ExtraData    argBytes       `json:"extraData"`
-	MixHash      types.Hash     `json:"mixHash"`
-	Nonce        types.Nonce    `json:"nonce"`
-	Hash         types.Hash     `json:"hash"`
-	Transactions []*transaction `json:"transactions"`
+	ParentHash      types.Hash     `json:"parentHash"`
+	Sha3Uncles      types.Hash     `json:"sha3Uncles"`
+	Miner           types.Address  `json:"miner"`
+	StateRoot       types.Hash     `json:"stateRoot"`
+	TxRoot          types.Hash     `json:"transactionsRoot"`
+	ReceiptsRoot    types.Hash     `json:"receiptsRoot"`
+	LogsBloom       types.Bloom    `json:"logsBloom"`
+	Difficulty      argUint64      `json:"difficulty"`
+	TotalDifficulty argUint64      `json:"totalDifficulty"`
+	Size            argUint64      `json:"size"`
+	Number          argUint64      `json:"number"`
+	GasLimit        argUint64      `json:"gasLimit"`
+	GasUsed         argUint64      `json:"gasUsed"`
+	Timestamp       argUint64      `json:"timestamp"`
+	ExtraData       argBytes       `json:"extraData"`
+	MixHash         types.Hash     `json:"mixHash"`
+	Nonce           types.Nonce    `json:"nonce"`
+	Hash            types.Hash     `json:"hash"`
+	Transactions    []*transaction `json:"transactions"`
+	Uncles          []*uncle       `json:"uncles"`
 }
 
 func toBlock(b *types.Block) *block {
 	h := b.Header
 	res := &block{
-		ParentHash:   h.ParentHash,
-		Sha3Uncles:   h.Sha3Uncles,
-		Miner:        h.Miner,
-		StateRoot:    h.StateRoot,
-		TxRoot:       h.TxRoot,
-		ReceiptsRoot: h.ReceiptsRoot,
-		LogsBloom:    h.LogsBloom,
-		Difficulty:   argUint64(h.Difficulty),
-		Number:       argUint64(h.Number),
-		GasLimit:     argUint64(h.GasLimit),
-		GasUsed:      argUint64(h.GasUsed),
-		Timestamp:    argUint64(h.Timestamp),
-		ExtraData:    argBytes(h.ExtraData),
-		MixHash:      h.MixHash,
-		Nonce:        h.Nonce,
-		Hash:         h.Hash,
-		Transactions: []*transaction{},
+		ParentHash:      h.ParentHash,
+		Sha3Uncles:      h.Sha3Uncles,
+		Miner:           h.Miner,
+		StateRoot:       h.StateRoot,
+		TxRoot:          h.TxRoot,
+		ReceiptsRoot:    h.ReceiptsRoot,
+		LogsBloom:       h.LogsBloom,
+		Difficulty:      argUint64(h.Difficulty),
+		TotalDifficulty: argUint64(h.Difficulty), // not needed for POS
+		Size:            argUint64(0),            // should derive actual size
+		Number:          argUint64(h.Number),
+		GasLimit:        argUint64(h.GasLimit),
+		GasUsed:         argUint64(h.GasUsed),
+		Timestamp:       argUint64(h.Timestamp),
+		ExtraData:       argBytes(h.ExtraData),
+		MixHash:         h.MixHash,
+		Nonce:           h.Nonce,
+		Hash:            h.Hash,
+		Transactions:    []*transaction{},
+		Uncles:          []*uncle{},
```

Comment:
If you accept the previous comment here we can do something like this (not sure if I got the alignment correct, github can't format 🙂):
```suggestion
		uncle:           *toUncle(h),
		Transactions:    []*transaction{},
		Uncles:          []*uncle{},
```

Reply:
Done. Thanks for the suggestion @Vuksan 👍